core/primitives/src/crypto.rs

```diff
@@ -452,6 +452,41 @@ pub mod key_types {
 	pub const SR25519: KeyTypeId = 20;
 }
 
+/// An enum of known key types.
+#[cfg(feature = "std")]
+pub enum KeyType {
```

Comment:
the reason I didn't set this out in the first place is that we don't want to claim to define _all_ key types. It's better to do the typedef because it gives users freedom to define their own crypto.